### PR TITLE
Issue 4859: Interactive Demo Tool

### LINKED ISSUE #4859: Interactive Demo Tool
**Problem description**
Create an interactive demo tool that uses the Pravega Client to execute various Scope/Stream/KVT operations on Pravega.

**Problem location**
Interactive Demo.

**Suggestions for an improvement**
Make the tool.

### PR DESCRIPTION
**Pull request by andreipaduroiu**
**Change log description**  
- Initial implementation of Pravega's IDT (Interactive Demo Tool). Supports Scopes, Streams and KVTs.

**Purpose of the change**  
Fixes #4859.

**What the code does**  
The IDT uses the Pravega Client to do the following operations:
- Scopes: create and delete.
- Streams: create, delete, list, append, read.
- KVTs: create, delete, put, get, remove, list keys/entries.

**How to verify it**  
1. Run standalone: `./gradlew startStandalone`
2. While standalone is running, run the tool in a different terminal:
    -  `./gradlew startInteractiveDemo -q --console=plain`
    - `-q` and `--console=plain` are required so that Gradle will get out of the way of console IO.

(Wiki page will be built [here](https://github.com/pravega/pravega/wiki/Interactive-Demo-Tool) probably after this PR is merged in).

### CODE REVIEW

**Comment by RaulGracia** [comment]
@andreipaduroiu as I see this PR, we could split the "chat" demo to be a PR in Pravega Samples. Here, we could merge the first part that may be candidate to get integrated in the Pravega CLI once we move it to this repo. What do you think?

**Comment by andreipaduroiu** [comment]
@RaulGracia That's a good idea. I removed `ChatService.java` from this PR. Please have a look.

I will open a PR against samples once 0.8 is published.

File: build.gradle [hunk 1]
```diff
@@ -552,6 +552,18 @@ project('test:integration') {
         standardInput = System.in
         systemProperties System.getProperties()
     }
+
+    /**
+     * See: https://github.com/pravega/pravega/wiki/Interactive-Demo-Tool .
```
Comment (RaulGracia) [inline build.gradle:557]: Nit: extra trailing space.

File: test/integration/src/main/java/io/pravega/test/integration/demo/interactive/ConfigCommand.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        Set(CommandArgs args) {
+            super(args);
+        }
+
+        @Override
+        public void execute() {
+            ensureMinArgCount(1);
+            val newValues = new HashMap<String, String>();
+            getCommandArgs().getArgs().forEach(s -> {
+                String[] items = s.split("=");
```
Comment (RaulGracia) [inline test/integration/src/main/java/io/pravega/test/integration/demo/interactive/ConfigCommand.java:41]: Perhaps the character `=` could be defined as a constant for readability.

File: test/integration/src/main/java/io/pravega/test/integration/demo/interactive/InteractiveDemo.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        System.out.println(String.format("%nType \"%s\" for list of commands, or \"%s\" to exit.", CMD_HELP, CMD_EXIT));
+        @Cleanup
+        Scanner input = new Scanner(System.in);
+        while (true) {
+            System.out.print(System.lineSeparator() + "> ");
+            String line = input.nextLine();
+            processCommand(line, config);
+        }
+    }
+
```
Comment (RaulGracia) [inline test/integration/src/main/java/io/pravega/test/integration/demo/interactive/InteractiveDemo.java:65]: Nit: too many blank lines.

File: test/integration/src/main/java/io/pravega/test/integration/demo/interactive/InteractiveDemo.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                break;
+            case CMD_EXIT:
+                System.exit(0);
+                break;
+            default:
+                execCommand(pc);
+                break;
+        }
+    }
+
```
Comment (RaulGracia) [inline test/integration/src/main/java/io/pravega/test/integration/demo/interactive/InteractiveDemo.java:85]: Nit: too many blank lines.

**Pull request merged**